lib/internal/fs/watchers.js

```diff
@@ -117,6 +121,21 @@ StatWatcher.prototype.stop = function() {
   this._handle = null;
 };
 
+StatWatcher.prototype.ref = function() {
+  // Avoid refCount calling ref multiple times causing unref to have no effect.
+  if (this[KFSStatWatcherRefCount] === this[KFSStatWatcherMaxRefCount])
+    return;
+  if (this._handle && this[KFSStatWatcherRefCount]++ === 0)
+    this._handle.ref();
+};
+
+StatWatcher.prototype.unref = function() {
+  // Avoid refCount calling unref multiple times causing ref to have no effect.
+  if (this[KFSStatWatcherRefCount] === 0) return;
+  if (this._handle && --this[KFSStatWatcherRefCount] === 0)
+    this._handle.unref();
+};
```

Comment:
For consistency with `ref()` and `unref()` used elsewhere throughout node.js (e.g. Timers, net, etc) ... these should both `return this`.